packages/firestore/test/unit/remote/bloom_filter.test.ts

```diff
@@ -0,0 +1,393 @@
+    const bloomFilter = new BloomFilter(
+      new Uint8Array([151, 153, 236, 116, 7]),
+      3,
+      13
+    );
+    expect(bloomFilter.size).to.equal(37);
+  });
+
+  it('should throw error if padding is negative', () => {
+    try {
+      new BloomFilter(new Uint8Array(1), -1, 1);
+      expect.fail();
+    } catch (error) {
+      expect(
+        (error as Error)?.message.includes(
+          'INTERNAL ASSERTION FAILED: Padding is negative.'
+        )
+      ).to.be.true;
+    }
+  });
+
+  it('should throw error if bitmap size is negative', () => {
+    try {
+      new BloomFilter(new Uint8Array(1), 9, 1);
+      expect.fail();
+    } catch (error) {
+      expect(
+        (error as Error)?.message.includes(
+          'INTERNAL ASSERTION FAILED: Bitmap size is negative.'
+        )
+      ).to.be.true;
+    }
+  });
+
+  it('should throw error if hash count is negative', () => {
+    try {
+      new BloomFilter(new Uint8Array(1), 1, -1);
+      expect.fail();
+    } catch (error) {
+      expect(
+        (error as Error)?.message.includes(
+          'INTERNAL ASSERTION FAILED: Hash count is 0 or negative'
+        )
+      ).to.be.true;
+    }
+  });
+
+  it('should throw error if hash count is 0 for non empty bloom filter', () => {
+    try {
+      new BloomFilter(new Uint8Array(1), 1, 0);
+      expect.fail();
+    } catch (error) {
+      expect(
+        (error as Error)?.message.includes(
+          'INTERNAL ASSERTION FAILED: Hash count is 0 or negative'
+        )
+      ).to.be.true;
+    }
+  });
+
+  it('mightContain in empty bloom filter should always return false', () => {
+    const bloomFilter = new BloomFilter(new Uint8Array(0), 0, 0);
+    expect(bloomFilter.mightContain('abc')).to.be.false;
+    expect(bloomFilter.mightContain('def')).to.be.false;
+  });
+
+  it('mightContain should always return false for empty string', () => {
+    const emptyBloomFilter = new BloomFilter(new Uint8Array(0), 0, 0);
+    const nonEmptyBloomFilter = new BloomFilter(
+      new Uint8Array([151, 153, 236, 116, 7]),
+      3,
+      13
+    );
+    expect(emptyBloomFilter.mightContain('')).to.be.false;
+    expect(nonEmptyBloomFilter.mightContain('')).to.be.false;
+  });
+
+  /**
+   * Golden tests are generated by backend based on inserting n number of
+   * documents path into a bloom filter.
+   *
+   * Full document path is generated by concating documentPrefix and number n,
+   * ie, projects/project-1/databases/database-1/documents/coll/doc12
+   *
+   * The test result is generated by checking the membership of documents from
+   * documentPrefix+0 to documentPrefix+2n. The membership results from 0 to n
+   * is expected to be true (1), and the membership results from n to 2n is
+   * expected to be false (0) with some false positive results (1).
+   *
+   * The test result of BloomFilter.mightContain() should match the backend
+   * result exactly.
+   */
+  describe('BloomFilter membership test', () => {
```

Comment:
There is a lot of repeated boilerplate logic in each of the golden test methods. Please factor out that repeated logic into a helper function. This function could probably even just take the input and expected result as arguments, making the individual tests pretty small.